### PR TITLE
chore(k8s): update Mutagen to v0.14.0

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:

This pull request updates the k8s-sync functionality to use Mutagen v0.14.0 (or, rather, Garden's Mutagen fork rebased atop Mutagen v0.14.0).  The [v0.14.0](https://github.com/mutagen-io/mutagen/releases/tag/v0.14.0) release brings some important optimizations to synchronization and filesystem watching.  This should hopefully be the last of the fork-based updates to Mutagen in Garden.

**Which issue(s) this PR fixes**:

No specific issues are fixed by this PR.

**Special notes for your reviewer**:
This reopens #2929 
CC @edvald @thsig

### CODE REVIEW

**Comment by vvagaytsev** [comment]
All broken e2e tests fail in the build step, i.e. while executing `garden build --force`. I'm wondering why the build command triggers the mutagen syncing. I disabled the build step in the e2e tests and everything went fine.

In the build step tests, the mutagen version downloaded to the garden project dirs is 0.14.0. It looks like the mutagen in `garden-dev-init` is out-of-date. I'll check the logic of the build task processing to figure out how and why the sync starts. To my understanding, the build step should not cause any mutagen syncing.

**Comment by vvagaytsev** [comment]
In https://github.com/garden-io/garden/pull/2955/commits/477711efab46f82eabc66f30942769975f9e068e I added a command to print the mutagen-agent version running in the target container (i.e. mutagen's beta endpoint).

The output can be found in the CircleCI integration tests' logs, for instance, [here](https://app.circleci.com/pipelines/github/garden-io/garden/11801/workflows/a4322613-4c1b-4010-9475-ae2184285705/jobs/187242).

If you search for "Mutagen version in" you'll see that containers running in dev mode have the correct up-to-date mutagen agent version, for example:
```
Mutagen version in dev-mode Deployment/dev-mode: Mutagen agent version 0.14.0
; verified by /home/circleci/.garden/tools/kubectl/49eb930aa565a80f/kubectl exec -i --context=minikube --namespace=container-default --container dev-mode Deployment/dev-mode -- /.garden/mutagen-agent --version
```
The dev mode uses the init container `garden-dev-init` based on the Docker image `gardendev/k8s-sync`. That ensures the up-to-date mutagen agent in the containers running in dev-mode.

The version checking command fails with the error for the following remote resources:
1. Container `util` with deployment name `Deployment/garden-buildkit`. This causes failures in integration tests.
2. Container `util` with deployment name `Deployment/garden-util`.  This causes failures in e2e tests.

Error is the same in both cases:
```
Error: unknown flag: --version
```

So, the `mutagen-agent` does not recognize the `--version` flag. It looks like the mutagen agent version is too old in these problematic containers. I need to figure out more details:
1. Images that are used for these containers. 
2. How is mutagen-agent getting copied/installed into these containers.

cc @edvald @thsig

**Comment by vvagaytsev** [comment]
Ok, as I mentioned in https://github.com/garden-io/garden/pull/2955#issuecomment-1132615196 the problem is that `Deployment/garden-buildkit` and `Deployment/garden-util` use the older version of `mutagen-agent` which is `0.13.0`. This can be verified from the [CI logs](https://app.circleci.com/pipelines/github/garden-io/garden/11814/workflows/97473b51-b741-436f-8ee9-ab952b742bda) by searching for text "mutagen version" in the page.

@edvald any ideas on how to fix it in a better way? It looks like this causes failure of all `cluster-buildkit` tests.

File: images/k8s-sync/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Image `k8s-sync`
+
+This image is used for code-synchronization in dev mode. It's based on the [Alpine Linux](https://www.alpinelinux.org/)
+and [Mutagen](https://github.com/mutagen-io/mutagen).
+
+## Mutagen version upgrade
+
+To update the mutagen version, two following steps must be done:
```
Comment (thsig) [inline images/k8s-sync/README.md:8]: "To update the mutagen version, two following steps must be done:" -> "To update the mutagen version:"

File: images/k8s-sync/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Image `k8s-sync`
+
+This image is used for code-synchronization in dev mode. It's based on the [Alpine Linux](https://www.alpinelinux.org/)
+and [Mutagen](https://github.com/mutagen-io/mutagen).
+
+## Mutagen version upgrade
```
Comment (thsig) [inline images/k8s-sync/README.md:6]: "Upgrade" -> "Update"

Same on lines 15 and 24.

**Review by thsig — CHANGES_REQUESTED** [review]
Looks good! Only had a few minor text-related comments there.

File: images/k8s-sync/README.md (hunk 2 above)
Comment (vvagaytsev) [reply images/k8s-sync/README.md]: Done

**Review by thsig — APPROVED** [review]

**Pull request merged**